frontend/src/firework.ts

```diff
@@ -0,0 +1,49 @@
+import Confetti from "canvas-confetti";
+var confetti: any = null;
+
+export const firework = (timeInSecond: number): void => {
+  if (confetti == null) {
+    confetti = Confetti.create(null, { resize: true });
+  }
+
+  var duration = timeInSecond * 1000;
+  var animationEnd = Date.now() + duration;
+  var defaults = { startVelocity: 30, spread: 360, ticks: 60, zIndex: 0 };
+  var handler: any = null;
+
+  function randomInRange(min: number, max: number): number {
+    return Math.random() * (max - min) + min;
+  }
+
+  var interval: number;
+
+  interval = setInterval(function () {
+    var timeLeft = animationEnd - Date.now();
+
+    if (timeLeft <= 0) {
+      handler != null && window.removeEventListener("click", handler);
+      return clearInterval(interval);
+    }
+
+    var particleCount = 300 * (timeLeft / duration);
+    // since particles fall down, start a bit higher than random
+    confetti(
+      Object.assign({}, defaults, {
+        particleCount,
+        origin: { x: randomInRange(0.1, 0.3), y: Math.random() - 0.2 },
+      })
+    );
+    confetti(
+      Object.assign({}, defaults, {
+        particleCount,
+        origin: { x: randomInRange(0.7, 0.9), y: Math.random() - 0.2 },
+      })
+    );
+  }, 250);
+
+  setTimeout(() => {
+    handler = window.addEventListener("click", (event) => {
```

Comment:
This will leak event listeners on the Window, I think? Since there's no guarantee that we'll re-run this function after `clearInterval` is called.

I'd ~strongly prefer that this is rewritten as a React component

Reply:
You are right. Will fix. The listener is meant for players to stop the firework before it runs its cause.